### PR TITLE
ddl: recover panic in add index workers

### PR DESCRIPTION
**Pull request by winkyao**
Recover panic in add index workers, and return task with error normally.

### CODE REVIEW

**Comment by jackysp** [comment]
LGTM

File: ddl/index.go [hunk 1]
```diff
@@ -613,6 +613,16 @@ func (w *addIndexWorker) handleBackfillTask(task *reorgIndexTask) *addIndexResul
 }
 
 func (w *addIndexWorker) run() {
+	log.Infof("[ddl-reorg] worker[%v] start", w.id)
+	defer func() {
+		r := recover()
+		if r != nil {
+			buf := util.GetStack()
+			log.Errorf("addIndexWorker %v %s", r, buf)
```
Comment (zimulala) [inline ddl/index.go]: Add `[ddl-reorg]` to the log?

**Comment by coocood** [comment]
LGTM

**Comment by winkyao** [comment]
@zimulala Please take a look

File: ddl/index.go [hunk 2]
```diff
@@ -613,6 +613,16 @@ func (w *addIndexWorker) handleBackfillTask(task *reorgIndexTask) *addIndexResul
 }
 
 func (w *addIndexWorker) run() {
+	log.Infof("[ddl-reorg] worker[%v] start", w.id)
+	defer func() {
+		r := recover()
+		if r != nil {
+			buf := util.GetStack()
+			log.Errorf("[ddl-reorg] addIndexWorker %v %s", r, buf)
+			metrics.PanicCounter.WithLabelValues(metrics.LabelDDL).Inc()
```
Comment (shenli) [inline ddl/index.go]: Add a new label named ddl-addindex-worker?

Comment (winkyao) [reply ddl/index.go]: I think metrics.LabelDDL is enough.

**Pull request merged**